### DISCUSSION TITLE
Using turborepo to get a consistent value for NextJS generateBuildId()

### DISCUSSION THREAD

**Discussion by VanTanev** [vercel/turborepo#7154] [Help] [answered]
We're running a monorepo that includes a couple of nextjs projects. Those are deployed to AWS as docker containers.

Unfortunately, `next build` generates random `.next/BUILD_ID` by default, which means disparate builds of the same container result in different containers being generated.

One suggested workaround is to use:
```ts
// next.config.js
module.exports = {
  generateBuildId: async () => {
    // This could be anything, using the latest git hash
    return process.env.GIT_HASH
  },
}
```

This however is too coarse for a monorepo, where nextjs builds will usually not be affected by most commits.

Is there a way to get the turbo hash for a particular command + workspace combination (eg, `turbo run build --filter=my-nextjs-workspace`), and somehow plug that in, to get consistent build output that generates into the same Docker container?

Or, if anyone has ran into this, other workaround suggestions would be most welcome.

**Comment by mehulkar** [comment, marked as answer]
`turbo run build --filter=my-nextjs-workspace --dry=json` will give you the hash of the app. You can filter out the hash with a JSON parsers or by piping into `| jq` on the CLI!

**Reply by VanTanev** [reply]
This is EXACTLY what I was looking for, but when I looked at the documentation I didn't find it. I guess I expected it would be a separate command instead of an option to `run`, but the interface makes sense. Thanks for pointing it out!

**Reply by mehulkar** [reply]
It's good feedback that this could have more prominence in our docs, even if we don't make it a separate command. cc @anthonyshew to keep tabs on this as he works on the docs!
